modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -785,48 +786,382 @@ public void testGetPaginatedOrganizationsWithDefaultLimit() {
+    public void testGetPaginatedOrganizationsDiscoveryForNumericEdgeCasesOfLimit(int offset, int limit) {
+
+        String queryUrl = buildQueryUrl(offset, limit);
+
+        Response response = getResponseOfGetWithOAuth2(queryUrl, m2mToken);
+
+        validateHttpStatusCode(response, HttpStatus.SC_OK);
+
+        // Validate the response content
+        int actualCount = response.jsonPath().getInt(COUNT_PATH_PARAM);
+        int totalResults = response.jsonPath().getInt(TOTAL_RESULTS_PATH_PARAM);
+        int startIndex = response.jsonPath().getInt(START_INDEX_PATH_PARAM);
+        List<Map<String, String>> links = response.jsonPath().getList(LINKS_PATH_PARAM);
+        List<Map<String, String>> returnedOrganizations = response.jsonPath().getList("organizations");
+
+        int expectedCount = Math.min(limit, (NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS - offset));
+
+        Assert.assertEquals(actualCount, expectedCount,
+                "Unexpected number of organizations returned for limit: " + limit);
+        Assert.assertEquals(totalResults, NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS,
+                "Total results should match the number of organizations available.");
+        Assert.assertEquals(startIndex, offset + 1,
+                "Start index should always be 1 greater than the offset.");
+
+        validateOrganizationDiscoveryLimitEdgeCaseLinks(links, limit, offset);
+        validateOrganizationDiscoveryLimitEdgeCaseOrganizations(returnedOrganizations, limit, offset);
+    }
+
+    @Test(groups = "organizationDiscoveryPaginationTests", dependsOnMethods = "testAddEmailDomainsToOrganization")
+    public void testGetPaginatedOrganizationsDiscoveryForNonNumericEdgeCasesOfLimit() {
+
+        // Test case 1: URL with LIMIT_QUERY_PARAM but no value.
+        String endpointURLWithEmptyLimit =
+                ORGANIZATION_MANAGEMENT_API_BASE_PATH + ORGANIZATION_DISCOVERY_API_PATH + QUESTION_MARK +
+                        FILTER_QUERY_PARAM + EQUAL + AMPERSAND + LIMIT_QUERY_PARAM + EQUAL + AMPERSAND +
+                        OFFSET_QUERY_PARAM + EQUAL + ZERO;
+
+        Response responseWithEmptyLimit = getResponseOfGetWithOAuth2(endpointURLWithEmptyLimit, m2mToken);
+        validateHttpStatusCode(responseWithEmptyLimit, HttpStatus.SC_OK);
+
+        validateResponseForOrganizationDiscoveryLimitDefaultCases(responseWithEmptyLimit);
+
+        // Test case 2: URL without LIMIT_QUERY_PARAM.
+        String endpointURLWithoutLimit =
+                ORGANIZATION_MANAGEMENT_API_BASE_PATH + ORGANIZATION_DISCOVERY_API_PATH + QUESTION_MARK +
+                        FILTER_QUERY_PARAM + EQUAL + AMPERSAND + OFFSET_QUERY_PARAM + EQUAL + ZERO;
+
+        Response responseWithoutLimit = getResponseOfGetWithOAuth2(endpointURLWithoutLimit, m2mToken);
+        validateHttpStatusCode(responseWithoutLimit, HttpStatus.SC_OK);
+
+        validateResponseForOrganizationDiscoveryLimitDefaultCases(responseWithoutLimit);
```

Comment:
You can split this into two test methods

Reply:
Addressed with commit: https://github.com/wso2/product-is/pull/21023/commits/6ce4d2238cbc4e98ef8071dd6fa6ec098d6f9ba2